FrontEnd/src/App.jsx

```diff
@@ -39,6 +39,31 @@ function App() {
     fetchData();
   }, [userUID]);
 
+  useEffect(() => {
+    const newSocket = io("http://localhost:5000");
+    setSocket(newSocket);
+
+    newSocket.on("connect", () => {
+      console.log("Connected to server with socket ID:", newSocket.id);
+    });
+
+    newSocket.on("disconnect", (reason) => {
+      console.log("Disconnected from server:", reason);
+    });
+
+    newSocket.on("reconnect", (attemptNumber) => {
+      console.log("Reconnected to server after", attemptNumber, "attempts");
+    });
+
+    newSocket.on("firstEvent", (msg) => {
+      console.log(msg);
```

Comment:
You'll remove these `console.log`s later?

Reply:
Yes, I did in my later commit